### ISSUE TITLE
 vNIC in Windows container suddenly disappears

### ISSUE THREAD

**Issue by rkttu-devsisters** [kubernetes/kubernetes#65163]
**Is this a BUG REPORT or FEATURE REQUEST?**:

/kind bug
@kubernetes/sig-windows-bugs

**What happened**:
- I first deployed the Microsoft IIS Pod (server version 1803) via a YAML file.
- Immediately after the distribution, I checked the remote shell via kubectl exec command to normally send and receive HTTP requests between Linux Pod and Windows Pod.
- I do not know what happens after a certain amount of time, but suddenly the network adapter inside the Pod disappears from the Pod if I run the Pod with no HTTP request for more than one day.
- Compartment details of the host also indicate that the adapter has disappeared.
- In this state, I erase the corresponding pod and re-distribute it, and the network adapter is connected to the inside of the Pod again.

**What you expected to happen**:
Container/Pod shoult not lost their virtual network adapter.

**How to reproduce it (as minimally and precisely as possible)**:
I apologize for the difficulty in reproducing the issue that occurs. It seems to happen if no HTTP request arrives for about a day or so.

**Anything else we need to know?**:
I do not know if this is helpful, but I also attach a log file that documents the situation at the time the problem occurred.
[iis-1803-diagnose-log.zip](https://github.com/kubernetes/kubernetes/files/2107974/iis-1803-diagnose-log.zip)

**Environment**:
- Kubernetes version (use `kubectl version`): 1.9.3
- Cloud provider or hardware configuration: AWS (with ENI enhancements)
- OS (e.g. from /etc/os-release): Created custom AMI from official Windows Server 1803 AMI (Windows_Server-1803-English-Core-ContainersLatest-2018.05.09, ID: ami-dcc822a3)
- Kernel (e.g. `uname -a`): 64bit, 10.0.17134.1
- Install tools: Existing K8S cluster installed with KOPS, Windows node was installed manually
- Joined into existing Kubernetes 1.9.3 cluster (Existing cluster built with KOPS, Network Model: kubenet)
- I installed Windows node networking with this repo: https://github.com/Microsoft/SDN/tree/master/Kubernetes/windows
- Docker version: 17.06.2-ee-7 (amd64)
- All external network requests pass through NGINX ingress controller. (No node-port or AWS ELB used)

**Comment by rkttu-devsisters** [comment]
I checked the event log on the host computer and found that there were some repeated messages since joining the Kubernetes cluster and running the IIS pod.

- "The operation \u0027Get Stats\u0027 failed on nic 569AF199-B204-4C5D-B930-6D150E99187A (Friendly Name: ). Status = {Buffer Too Small}\r\nThe buffer is too small to contain the entry. No information has been written to the buffer.."
- "Failed to get Device VlanID Regkey for Physical NIC /DEVICE/{E6578F92-62EE-4B74-8A13-DC27C7763945}. vSwitch will be created on this Physical NIC without VlanID being inherited by HostVNic. Status = Object Name not found."
- "The disk signature of disk 8 is equal to the disk signature of disk 6."
- "Disk 8 has the same disk identifiers as one or more disks connected to the system. Go to Microsoft\u0027s support website (http://support.microsoft.com) and search for KB2983588 to resolve the issue."

Messages were mainly about virtual network adapters or virtual disks. In particular, for virtual network adapters, it appears that a bug appears in which the network adapter disappears shortly after the associated event log occurs.

I will attach an additional event log file that I collected.

[event-logs-k8s-worker-1803.zip](https://github.com/kubernetes/kubernetes/files/2108307/event-logs-k8s-worker-1803.zip)

**Comment by rkttu-devsisters** [comment]
I suspect that the version of the node binaries might be out of date, so I'm testing whether the problem can be reproduced by upgrading only the Windows node to 1.10.4, apart from the version 1.9.3 of the existing cluster.

The same situation is relatively easy to reproduce, so I will review the same symptoms after the upgrade and comment on this issue.

**Comment by rkttu-devsisters** [comment]
After the upgrade, I get the same symptoms. :-(

**Comment by rkttu-devsisters** [comment]
While examining kubelet's log file, I think I found the cause of the problem, so I leave an additional post.

Looking at the contents of the log file, WinCNI attempted to delete the HNS endpoint as shown below.

```json
{"level":"debug","msg":"[cni-net] Processing DEL command with args {ContainerID:a264985aca70ad3755cf3993f5fba31278fca33c9b2955652149b0bae6b3850e Netns:none IfName:eth0 Args:IgnoreUnknown=1;K8S_POD_NAMESPACE=default;K8S_POD_NAME=iis2;K8S_POD_INFRA_CONTAINER_ID=a264985aca70ad3755cf3993f5fba31278fca33c9b2955652149b0bae6b3850e Path:c:\\k\\cni}.","time":"2018-06-17T06:28:45Z"}
```

So I followed the contents of this log. When I removed the HNS Endpoint from the host that I associated with my IIS Deployed Pod, I experienced the same symptoms as this issue.

ipmo C:\k\hns.psm1
$test = Get-HnsEndpoint | Where {$_.Name -eq "581e9a0ad59415084a6b7dcd334bc961cc1948b5eeb36f6ac0d7fb56d761602e_l2bridge"}
Remove-HnsEndpoint $test

If so, I wonder why WinCNI suddenly deleted the HNS endpoint of the container in good condition. I am waiting for this phenomenon to reappear, and this time I will look back at the kubelet log by verifying the exact name of the HNS Endpoint.

The location of the source code I found with the error message recorded in the log is as follows. Is anyone aware of this problem? Could you give me some explanation? Here are the locations I found:

https://github.com/Microsoft/SDN/blob/6b9387b44fcba9b2606251c534f7b4fa31ebde11/Kubernetes/wincni/cni/network/network.go

**Comment by rkttu-devsisters** [comment]
Since I first posted the issue, I have been thinking about various troubleshooting methods. However, there is a way to solve the problem that is really absurd. Fixed issue where vNIC suddenly disappears from container after adding Liveness Check to Pod. It seems to be independent of the way of liveness check (eg HTTP, TCP, Exec). It usually took 4 hours, but now it will not run for more than 1 day.

There is no way to imagine how Liveness Check is related to the automatic removal of a vNIC. I need a detailed explanation of this issue. It seems that Linux pod does not have to do this either.

**Comment by rkttu-devsisters** [comment]
I think the issue is not covered, do I have to more information to provide? Or is this not a bug?

**Comment by rkttu** [comment]
@PatrickLang I am still waiting for the answer to this issue. Is this a problem that needs to be solved by upgrading to the latest version of Kubernetes? If the problem is solved by upgrading to the latest version, I would like to close the issue by selecting Pod liveness check as workaround.

**Comment by PatrickLang** [comment]
cc @daschott 

There is an issue under investigation where if the last pod is stopped, HNS endpoints can be deleted. I'm wondering if this is related. A fix is in the works for that issue on Windows Server version 1803

The workaround for that is to create a daemon set to keep a pod running before you start your other pod that's associated with a service.

**Comment by rkttu-devsisters** [comment]
@PatrickLang It looks similar to the problem you mentioned.

However, if you run another IIS pod that does not add a pod liveness check, the HNS endpoint of the newly added pod has been deleted. In contrast, all pods running with pod liveness check retain their HNS endpoints. Note that the Kubernetes cluster already has an IIS pod that I have added to the Pod liveness check before and it is running for more than 5 days without the HNS endpoint being deleted.

The problem occurs exactly when I did not add a pod liveness check.

**Comment by rkttu-devsisters** [comment]
@PatrickLang @daschott The issue has been open for a long time, and I want to know if anything has been done.

**Comment by daschott** [comment]
Thank you for the detailed logs, this is helpful. I tried to reproduce this issue on 1.10.4 on-prem but all vNICs stayed... Where in the Windows CNI repository does it say that you need flannel? You can use flannel for automated route management, but you can also do this manually.

Either way, this issue is on our radar and we will follow up on this as soon as possible.

**Comment by rkttu-devsisters** [comment]
I think this is something I did not mention, so I leave more comments. The current version of Windows Server 1803 is version 10.0.17134.1. Will upgrading to a newer version of Windows Server 1803 resolve the problem?

**Comment by rkttu-devsisters** [comment]
@PatrickLang @daschott I've built it in another environment, but this time with the HTTP pod liveness check, there is a constant restart in the Pod. After the vNIC is randomly deleted according to the situation, it seems that the liveness check algorithm fails and the pod is continuously rebuilt. The conclusion seems to be the problem of WINCNI.

```
NAME                                     READY     STATUS    RESTARTS   AGE
external-dns-78d4584f79-zt767            1/1       Running   0          122d
iis                                      1/1       Running   25         2d
iis4                                     1/1       Running   2          2d
ingress-nginx-internal-f65b8bf96-m6gt9   1/1       Running   0          122d
ingress-nginx-office-79c9b4f65f-5njjk    1/1       Running   0          122d
ingress-nginx-public-c57b6576b-8rk28     1/1       Running   1          122d
log-test-server-8967f4898-lvg92          1/1       Running   0          16d
nginx-default-backend-5ff76b7d8-xwzxg    1/1       Running   0          122d
stove-784f88f555-5qs2b                   1/1       Running   0          3d
```

**Comment by rkttu-devsisters** [comment]
@PatrickLang @daschott I recently tested after upgrading to Windows Server 1803 version 17134.167, the frequency at which vNICs in the container appear to be deleted has changed. In other words, the problem is still not resolved.

About two or three hours after I put the Pod on, the problem appeared, so I looked at the kubelet log. You will see a message to disconnect the Pod's network as shown below. (IIS1 is the pod which has an issue.)

```
I0718 19:10:15.250256    3372 config.go:99] Looking for [api], have seen map[api:{}]
I0718 19:10:15.250256    3372 kubelet.go:1927] SyncLoop (housekeeping)
I0718 19:10:15.256256    3372 cni_windows.go:47] GetPodNetworkStatus result IP4:{IP:{IP:100.96.44.187 Mask:ffffff00} Gateway:100.96.44.2 Routes:[]}, DNS:{Nameservers:[] Domain: Search:[] Options:[]}

I0718 19:10:15.256256    3372 plugins.go:426] Calling network plugin cni to tear down pod "iis1_default"

I0718 19:10:15.258257    3372 cni.go:284] Got netns path none
I0718 19:10:15.258257    3372 cni.go:285] Using podns path default
I0718 19:10:15.258257    3372 cni.go:274] About to del CNI network l2bridge (type=wincni.exe)
```

In the case of a pod with a pod liveness check added, the vNIC is not deleted even after a few hours. I want to know why the vNIC is being removed by showing this log in 2-3 hours after the pod has been added to the pod without adding pod liveness check.

In other words, I want to know when and why the code that leaves such logs is called. And I want to know if I must include liveness check to avoid this situation. The version of the cluster is 1.9.3, and the version of the Windows worker node is 1.9.4.

For your help, I also attach a full kubelet log file for the section where the problem occurred.

[kubelet.err-20180718T110556.297.zip](https://github.com/kubernetes/kubernetes/files/2206531/kubelet.err-20180718T110556.297.zip)

**Comment by daschott** [comment]
I'm afraid I can't get a repro of this, neither on-prem nor on Azure using acs-engine (albeit this uses Azure CNI and this sounds like wincni issue). I will keep trying.

@rkttu-devsisters all I need to repro is setup WS1803 and schedule an IIS pod, correct?

**Comment by rkttu-devsisters** [comment]
@daschott Yes, but I don't think there will be any problems with the Kubernetes cluster built with acs-engine. I think you've already done a lot of testing with acs-engine. As I mentioned when I first started this issue, the problem I'm having is a Kubernetes cluster that I built with AWS EC2 using KOPS, not Azure.

Can you test in the same environment? If you can do that, I'll give you more configuration information about how I installed it.

**Comment by rkttu-devsisters** [comment]
@PatrickLang @daschott I tried changing the CNI plugin to Flannel (comes from Microsoft SDN repo) and enabled L2Bridge, but the problem is the same. I guess CNI plugin is not the cause of the problem. It seems to be the question of when to call the CNI plugin, or when HNS will randomly remove the container's vNIC.

**Comment by rkttu-devsisters** [comment]
@PatrickLang @daschott 

After running the container for more than one day, the problem was repeated again. The vNIC on the Windows Pod of another node that was not affected by an abnormal termination of the node was arbitrarily deleted.

Unfortunately, besides the problems I knew at the beginning, sometimes BSoD happens.

It is difficult to reproduce the problem, but in such a situation it is difficult to apply the Kubernetes Windows node in practice even for internal testing. In an environment other than Azure, I would like to know if it is impossible to officially build a Kubernetes Windows worker node.

**Comment by rkttu-devsisters** [comment]
@PatrickLang @daschott 

Perhaps this is the last information I can add. Unlike Kubernetes clusters, which are built in a clean state, existing deployments are failing to deploy Linux Pod to Windows nodes through Daemonset, regardless of platform. It is roughly the same situation as the log below.

```
C:\Users\rkttu>kubectl get pods --all-namespaces
NAMESPACE             NAME                                                                        READY     STATUS             RESTARTS   AGE
ck-server             cockroachdb-7db5985c4d-x44ln                                                1/1       Running            0          28d
crjw-dev-google0622   cb-5966c7455d-wwb9z                                                         1/1       Running            0          42d
crjw-dev-google0622   chatserver-0-5fc6bb4bcc-qcbhs                                               1/1       Running            0          42d
crjw-dev-google0622   gameserver-7b976d7886-99mzz                                                 1/1       Running            0          42d
crjw-dev-google0622   mysql-7b56c96577-xlrk5                                                      1/1       Running            0          42d
crjw-dev-google0622   pvpserver-0-5dd86fdc97-2t6tl                                                1/1       Running            0          42d
crjw-dev-google0622   redis-74fd46665c-r4m9v                                                      1/1       Running            0          42d
crjw-dev-google0622   web-terminal-7b8f867b96-cjd4x                                               0/1       ImagePullBackOff   0          42d
default               external-dns-78d4584f79-zt767                                               1/1       Running            0          141d
default               iis1                                                                        1/1       Running            0          1d
default               iis2                                                                        1/1       Running            0          19h
default               iis3                                                                        1/1       Running            0          1d
default               iis4                                                                        1/1       Running            0          1d
default               ingress-nginx-internal-f65b8bf96-m6gt9                                      1/1       Running            0          141d
default               ingress-nginx-office-79c9b4f65f-5njjk                                       1/1       Running            0          141d
default               ingress-nginx-public-c57b6576b-8rk28                                        1/1       Running            1          141d
default               log-test-server-8967f4898-lvg92                                             1/1       Running            0          35d
default               nginx-default-backend-5ff76b7d8-xwzxg                                       1/1       Running            0          141d
default               stove-784f88f555-5qs2b                                                      1/1       Running            0          22d
kube-system           dns-controller-756b8cd8-dt7cl                                               1/1       Running            0          143d
kube-system           etcd-server-events-ip-172-22-254-114.ap-northeast-1.compute.internal        1/1       Running            4          143d
kube-system           etcd-server-ip-172-22-254-114.ap-northeast-1.compute.internal               1/1       Running            4          143d
kube-system           filebeat-57rhf                                                              0/1       ImagePullBackOff   0          17d
kube-system           filebeat-5xg77                                                              1/1       Running            8          35d
kube-system           filebeat-d225x                                                              0/1       ImagePullBackOff   0          23d
kube-system           filebeat-f9qz9                                                              0/1       ImagePullBackOff   0          11d
kube-system           filebeat-ph8zk                                                              0/1       ImagePullBackOff   0          1d
kube-system           filebeat-q5sqn                                                              1/1       Running            7          35d
kube-system           heapster-597bcd95b4-nd7vn                                                   2/2       Running            0          141d
kube-system           kube-apiserver-ip-172-22-254-114.ap-northeast-1.compute.internal            1/1       Running            4          143d
kube-system           kube-controller-manager-ip-172-22-254-114.ap-northeast-1.compute.internal   1/1       Running            0          143d
kube-system           kube-dns-56cf5cb57d-5tcz8                                                   3/3       Running            0          143d
kube-system           kube-dns-56cf5cb57d-bhgjq                                                   3/3       Running            0          143d
kube-system           kube-dns-autoscaler-787d59df8f-g9g52                                        1/1       Running            0          143d
kube-system           kube-proxy-ip-172-22-254-114.ap-northeast-1.compute.internal                1/1       Running            0          143d
kube-system           kube-proxy-ip-172-22-254-172.ap-northeast-1.compute.internal                1/1       Running            0          143d
kube-system           kube-proxy-ip-172-22-254-99.ap-northeast-1.compute.internal                 1/1       Running            0          143d
kube-system           kube-scheduler-ip-172-22-254-114.ap-northeast-1.compute.internal            1/1       Running            0          143d
kube-system           kubernetes-dashboard-5bd6f767c7-wnnlq                                       1/1       Running            0          141d
sdkdemo-t2            mysql-5c976648d7-pn7lb                                                      1/1       Running            0          81d
sdkdemo-t2            redis-776cbbc589-2tdsp                                                      1/1       Running            0          81d
sdkdemo-t2            sdkdemo-59887cf754-qxh2q                                                    1/1       Running            0          81d
sdkdemo-t2            web-terminal-7474bf9995-x2pz5                                               0/1       ImagePullBackOff   0          81d
sdktest-coupon2       mysql-5844886759-7hwqs                                                      1/1       Running            0          23d
sdktest-coupon2       redis-68c8459ccd-x2tvm                                                      1/1       Running            0          136d
sdktest-coupon2       sdktest-6798b4878c-dkffd                                                    1/1       Running            0          23d
sdktest-coupon2       web-terminal-5c98668885-2bnsh                                               0/1       ImagePullBackOff   0          136d
sdktest-coupon3       mysql-6b8f7b758f-jwvh5                                                      1/1       Running            0          135d
sdktest-coupon3       redis-55bc7d9469-rdzxs                                                      1/1       Running            0          135d
sdktest-coupon3       sdktest-84544f8868-2d7z5                                                    1/1       Running            0          135d
sdktest-coupon3       web-terminal-557d65dc59-mldg8                                               0/1       ImagePullBackOff   0          135d
sdktest-test3-local   gameserver-cf7549d64-7m6hn                                                  1/1       Running            0          22d
sdktest-test3-local   redis-86d74cfff-crjcb                                                       1/1       Running            0          22d
```

As mentioned earlier, the vNIC is deleted from iis1 to iis4, and the running status of the pod is normal, but it can not be accessed from outside. And each Windows node has multiple Pods that have different platforms and failed to run.

It is just a naive assumption, however, that there is a bug in which BSOD occurs due to incorrect information being given to HNS in the presence of pods left in such unclear state, or a random deletion of a vNIC of a pod that has been successfully executed.

**Comment by rkttu-devsisters** [comment]
This problem still occurs when I test after upgrading from Windows Server 2016 AMI to Windows Server 2019 Insider Preview build 17733. The version of the Kubernetes node binaries used for testing is 1.10.3.

At this point both Kubernetes and Windows are heading towards GA, this issue is still not resolved. I will not know when the problem will be solved, and I will disseminate it to my colleagues and the community that they should use an alternative to apply the pod liveness check.

**Comment by rkttu** [comment]
@PatrickLang I keep watching that you're doing a lot of things to make the Kubernetes Windows version to GA state. Is there a bug related to this problem that you are experiencing during your failed E2E tests? If you have any updates about this issue, I would appreciate it if you share.

**Comment by daschott** [comment]
@rkttu I apologize for the delay in the investigation but we haven't tested Kubernetes on Windows on AWS much, and we do not see this issue reproduce anywhere else, which makes it difficult for us to investigate.

**Comment by rkttu** [comment]
@daschott If it helps, I am willing to give you all the data for the Windows EC2 instance that is experiencing the problem. Could you give me an intensive look at this problem?

**Comment by rkttu-devsisters** [comment]
@daschott To recapture this problem recently, I wrote about the unusual aspect of the event log that appeared after I reproduced the problem.

When I collected the logs of kubelet's info level and looked at them, I got an error message saying that the status of the pod was different from the one stored in the following way.

```
I1108 18:07:00.968542    3852 status_manager.go:554] Pod status is inconsistent with cached status for pod "iis_default(107c780a-e249-11e8-a5ce-069074dc8382)", a reconciliation should be triggered:
 {"phase":"Running","conditions":[{"type":"Initialized","status":"True","lastProbeTime":null,"lastTransitionTime":"2018-11-07T04:53:25Z"},{"type":"Ready","status":"

A: False","lastProbeTime":null,"lastTransitionTime":"2018-11-08T02:13:27Z"},{"type":"PodScheduled","status":"True","lastProbeTime":null,"lastTransitionTime":"2018-11-07T04:53:27Z"}],"hostIP":"172.22.254.144","podIP":"100.96.8.71","startTime":"2018-11-07T04:53:25Z","containerStatuses":[{"name":"iis","state":{"running":{"startedAt":"2018-11-08T02:13:25Z"}},"lastState":{},"ready":true,"restartCount":0,"image":"microsoft/iis:windowsservercore-1803","imageID":"docker-pullable://microsoft/iis@sha256:3975e5ea57523568914e5cbf8d2ba4b92c1272953767eb4d43827d886e83fc11","containerID":"docker://8092e6db9385f23d6da67c6496d249a494f57d102f850475234e9d3e103cf1cb"}],"qosClass":"Burstable"}

B: True","lastProbeTime":null,"lastTransitionTime":"2018-11-08T02:13:27Z"},{"type":"PodScheduled","status":"True","lastProbeTime":null,"lastTransitionTime":"2018-11-07T04:53:27Z"}],"hostIP":"172.22.254.144","podIP":"100.96.8.71","startTime":"2018-11-07T04:53:25Z","containerStatuses":[{"name":"iis","state":{"running":{"startedAt":"2018-11-08T02:13:25Z"}},"lastState":{},"ready":true,"restartCount":0,"image":"microsoft/iis:windowsservercore-1803","imageID":"docker-pullable://microsoft/iis@sha256:3975e5ea57523568914e5cbf8d2ba4b92c1272953767eb4d43827d886e83fc11","containerID":"docker://8092e6db9385f23d6da67c6496d249a494f57d102f850475234e9d3e103cf1cb"}],"qosClass":"Burstable"}

I1108 18:07:00.968542    3852 status_manager.go:440] Status Manager: syncPod in syncbatch. pod UID: "107c780a-e249-11e8-a5ce-069074dc8382"
I1108 18:07:00.973539    3852 round_trippers.go:406] GET https://api.kubernetes.devsisters.systems/api/v1/namespaces/default/pods/iis 200 OK in 4 milliseconds
I1108 18:07:00.977538    3852 config.go:297] Setting pods for source api
I1108 18:07:00.978539    3852 kubelet.go:1878] SyncLoop (RECONCILE, "api"): "iis_default(107c780a-e249-11e8-a5ce-069074dc8382)"
I1108 18:07:00.980540    3852 round_trippers.go:406] PUT https://api.kubernetes.devsisters.systems/api/v1/namespaces/default/pods/iis/status 200 OK in 7 milliseconds
I1108 18:07:00.980540    3852 status_manager.go:481] Status for pod "iis_default(107c780a-e249-11e8-a5ce-069074dc8382)" updated successfully: (2, {Phase:Running Conditions:[{Type:Initialized Status:True LastProbeTime:0001-01-01 00:00:00 +0000 UTC LastTransitionTime:2018-11-07 13:53:25 +0900 KST Reason: Message:} {Type:Ready Status:True LastProbeTime:0001-01-01 00:00:00 +0000 UTC LastTransitionTime:2018-11-08 11:13:27 +0900 KST Reason: Message:} {Type:PodScheduled Status:True LastProbeTime:0001-01-01 00:00:00 +0000 UTC LastTransitionTime:2018-11-07 13:53:27 +0900 KST Reason: Message:}] Message: Reason: NominatedNodeName: HostIP:172.22.254.144 PodIP:100.96.8.71 StartTime:2018-11-07 13:53:25 +0900 KST InitContainerStatuses:[] ContainerStatuses:[{Name:iis State:{Waiting:nil Running:&ContainerStateRunning{StartedAt:2018-11-08 11:13:25 +0900 KST,} Terminated:nil} LastTerminationState:{Waiting:nil Running:nil Terminated:nil} Ready:true RestartCount:0 Image:microsoft/iis:windowsservercore-1803 ImageID:docker-pullable://microsoft/iis@sha256:3975e5ea57523568914e5cbf8d2ba4b92c1272953767eb4d43827d886e83fc11 ContainerID:docker://8092e6db9385f23d6da67c6496d249a494f57d102f850475234e9d3e103cf1cb}] QOSClass:Burstable})
```

And when I looked at the event log of the Windows Server side node at the above point, event logs were recorded in the same sequence as the attached file.

[evtlogs.txt](https://github.com/kubernetes/kubernetes/files/2562005/evtlogs.txt)

I have not noticed before, but it looks like the event log just looks like shutting down the OS, restarting it, or deleting or deactivating the vNIC inside the container. Note that the container I'm testing currently is the `microsoft/iis:windowsservercore-1803` container (with process isolation).

I'm wondering if there is anything in the Windows container that can do this for itself.

The funny thing is, as I said before, if I add the pod liveness check to the YAML I want to distribute, this problem disappears. In addition, I wonder if the power settings affect the Docker container. I understand that using Windows Server Core OS does not cause those problem, but I am curious.

If possible, I would like to talk about this issue with the Windows Server team, and more precisely with the Windows container team.

**Comment by rkttu-devsisters** [comment]
@daschott I'm still testing, and I've taken the contents of https://raw.githubusercontent.com/Microsoft/SDN/master/Kubernetes/WebServer.yaml and deployed pod instead of IIS. Then, the problem described in this issue does not occur. (Container is running for at least 10 hours.)

This is interpreted as if IIS's application pool recycle causes problems associated with HNS endpoints, and I would like to further investigate other types of web server applications (eg, nodejs, .NET core, etc.).

**Comment by rkttu-devsisters** [comment]
@PatrickLang @daschott 

I thought this problem would be solved through the hotfix KB4467682 you mentioned at https://github.com/kubernetes/kubernetes/issues/66947, but that still did not solve the problem. Is it because the vNIC is deleted after a certain period of time because of the wrong OS's handling of Hairpin Mode?

And I tried building the Windows Node with the latest version of ACS Engine. So instead of using WinCNI and using Azure CNI instead, I wonder if WinCNI is now treated as a deprecated version.

**Comment by rkttu** [comment]
@daschott I tried to test whether this problem was solved by replacing WinCNI plug-in with WinBridge and HostLocal plug-in, as advised by Seattle last time, but this did not solve the problem. The current version of K8s I build is 1.10.7, the existing cluster uses kubenet as a network plugin, and the MTU setting is 9001.

**Comment by rkttu-devsisters** [comment]
I think this problem occurs if you do not specify the hostname-override parameter in the AWS environment the same as kubelet. Thanks to the Windows Container team for confirming this issue.

I will close the issue.

However, later in the Kubernetes installation guide, it would be better to add a detailed explanation with the troubleshooting guide etc. to see what problems can occur when this parameter is omitted. The correlation between the name of the parameter and this problem is hard to see.

**Comment by daschott** [comment]
Thanks for confirming, @rkttu-devsisters. I'll take note to add an entry to the troubleshooting section for the next docs update. Thank you for your patience, we hope that the next issue you report the community or container team can help with issues faster.
